### ISSUE TITLE
Discarding the prompt tokens only with the positive labels and not with the negative ones

### ISSUE THREAD

**Issue by javismiles** [xfactlab/orpo#32]
In these lines below,
why are we masking just the positive labels with the mask.logical_not() to discard the prompt tokens,
but we don't do the same with the negative labels? what's the reason to only do it with the positive ones? thank you very much

`        ### Discard the prompt tokens in NLL loss if true
        if self.disable_prompt_loss:
            mask = inputs['attention_mask'] * inputs['positive_attention_mask']
            pos_labels = pos_labels * mask.logical_not()
            pos_labels[pos_labels == 0] = self.pad
        ##################################################

        neg_labels[neg_labels == self.pad] = -100
        pos_labels[pos_labels == self.pad] = -100`

**Comment by jiwooya1000** [comment]
Hello @javismiles,

The attached lines intend to mask the loss of prompts while calculating the NLL loss, which is $L_{SFT}$ in the paper.
And $L_{SFT}$ is calculated for the chosen responses only.
For that reason, we are only handling the positive ones.

Hope it helps!

Thank you.

**Comment by javismiles** [comment]
yes that makes total sense, thank you :)
